### PR TITLE
fix(zentao): update zentao plugin, do lots of fixes

### LINKED ISSUE #5948: [Bug][zentao] Issues table and issue_repo_commits table less data
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

Collected "project type project", collected, but there is no bug 6119 6118 6117 6121 in the issues table, and there is no associated data between bug6119 6118 6117 and commit in the issue_repo_commits table

### What do you expect to happen

Data in the table is correct

### How to reproduce

img_v2_e0f65b65-028b-4ff5-a942-c59570a0c6eg
image
image

### Anything else

_No response_

### Version

v0.19

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by d4x1**
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [x] I have added relevant tests.
- [x] I have added relevant documentation.
- [x] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

### Summary
What does this PR do?

1. update zentao's bug collector,collect zentao's bugs by project and execution instead of product
2. fix projects' url #5960
3. add story_point when issue.type is story #5961
4. extract all task/bug/story ids from log comment #5948

### Does this close any open issues?
Closes #5948 #5960 #5961

### Screenshots
Include any relevant screenshots here.
image
image
image
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: backend/plugins/zentao/tasks/bug_collector.go [hunk 1]
```diff
@@ -32,23 +32,53 @@ const RAW_BUG_TABLE = "zentao_api_bugs"
...
+type collectBugInput struct {
+	Path        string
+	ProjectId   int64
+	ProductId   int64
+	ExecutionId int64
+}
+
 func CollectBug(taskCtx plugin.SubTaskContext) errors.Error {
 	data := taskCtx.GetData().(*ZentaoTaskData)
-	cursor, iterator, err := getProductIterator(taskCtx)
```
Comment (abeizn) [inline backend/plugins/zentao/tasks/bug_collector.go:37]: Is there really no need to get bugs from product anymore? Or it needs to be obtained from product, and it needs to be obtained from project？

Comment (d4x1) [reply backend/plugins/zentao/tasks/bug_collector.go:37]: maybe @Startrekzky   can help aswer it.

File: backend/plugins/zentao/tasks/shared.go [hunk 2]
```diff
@@ -179,29 +179,29 @@ func ignoreHTTPStatus404(res *http.Response) errors.Error {
...
-		return nil, nil, err
-	}
-	iterator, err := api.NewDalCursorIterator(db, cursor, reflect.TypeOf(input{}))
-	if err != nil {
-		cursor.Close()
-		return nil, nil, err
-	}
-	return cursor, iterator, nil
-}
+//func getProductIterator(taskCtx plugin.SubTaskContext) (dal.Rows, *api.DalCursorIterator, errors.Error) {
```
[the remark this answers is not in the document]
Comment (d4x1) [reply backend/plugins/zentao/tasks/shared.go:182]: I think we will use `getProductIterator ` later. Zentao plugin is under testing, and new requirements may oocur.

**Review by abeizn — APPROVED** [review]
lgtm

**Pull request merged**